test/gtest/ucp/test_ucp_ep_reconfig.cc

```diff
@@ -520,3 +523,92 @@ UCS_TEST_P(test_reconfig_asymmetric, resolve_remote_id)
 }
 
 UCP_INSTANTIATE_TEST_CASE_TLS(test_reconfig_asymmetric, shm_ib, "shm,ib");
+
+class test_reconfigure_update_rank : public test_reconfig_asymmetric {
+public:
+    void init() override
+    {
+        test_ucp_ep_reconfig::init();
+        modify_config("NUM_EPS", std::to_string(num_eps()));
+    }
+
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        add_variant_with_value(variants, UCP_FEATURE_TAG, 200, "mid");
+        add_variant_with_value(variants, UCP_FEATURE_TAG, 500, "large");
+    }
+
+    void create_entities_and_connect() override
+    {
+        create_entity(true, false);
+        create_entity(false, false);
+        sender().connect(&receiver(), get_ep_params());
+        receiver().connect(&sender(), get_ep_params());
+    }
+
+    unsigned num_eps() const
+    {
+        return get_variant_value();
+    }
+
+    void verify_configuration();
+};
+
+void test_reconfigure_update_rank::verify_configuration()
+{
+    auto r_sender   = static_cast<const entity*>(&sender());
+    auto r_receiver = static_cast<const entity*>(&receiver());
+
+    if (should_reconfigure()) {
+        EXPECT_TRUE(r_sender->is_reconfigured());
+        EXPECT_TRUE(r_receiver->is_reconfigured());

```

Comment:
do we check if the relevant transport (rc or dc) is selected?

Reply:
it's checked indirectly by checking that config was changed

Reply:
but how do we know it was changed in the right way? IMO better have something that checks the feature works in high level (for example, that with high ep count, some eps were promoted to rc after enough traffic was sent)
for sake of this test, can adjust the timeout/counts to make the promotion happen quicker.
how long does this test take to run?

Reply:
num_iteration was adjusted to a smaller number, so now tests run faster